conda/plugins/manager.py

```diff
@@ -651,6 +659,83 @@ def get_environment_specifier(
         else:
             return self.get_environment_specifier_by_name(source=source, name=name)
 
+    def get_environment_exporters(self) -> Iterable[CondaEnvironmentExporter]:
+        """
+        Yields all detected environment exporters.
+        """
+        yield from self.get_hook_results("environment_exporters")
+
+    def get_exporter_format_mapping(self) -> dict[str, CondaEnvironmentExporter]:
+        """
+        Get a mapping from format names (including aliases) to environment exporters.
+
+        :return: Dict mapping format name to CondaEnvironmentExporter
+        """
+        mapping = {}
+        seen_formats = set()
+
+        for plugin in self.get_environment_exporters():
+            for format_name in (plugin.name, *plugin.aliases):
+                if format_name in seen_formats:
+                    raise PluginError(
+                        f"Conflicting format name '{format_name}' found in environment exporters. "
+                        f"Multiple plugins cannot use the same format name or alias."
+                    )
+                seen_formats.add(format_name)
+                mapping[format_name] = plugin
+
+        return mapping
+
+    def detect_environment_exporter(
+        self, filename: str
+    ) -> CondaEnvironmentExporter | None:
+        """
+        Detect an environment exporter based on exact filename matching against default_filenames.
+
+        :param filename: Filename to find an exporter for (basename is used for detection)
+        :return: CondaEnvironmentExporter that supports the filename, or None if none found
+        :raises PluginError: If multiple exporters claim to support the same filename
+        """
+        # Extract just the basename for matching
+        basename = os.path.basename(filename)
+
+        matches = []
+        for exporter_config in self.get_environment_exporters():
+            # Check if basename exactly matches any of the default filenames
+            if basename in exporter_config.default_filenames:
+                matches.append(exporter_config)
+
+        if len(matches) == 1:
+            return matches[0]
+        elif len(matches) == 0:
+            return None  # No match found, return None instead of raising error
```

Comment:
I think it would be good for all the plugin manager's `detect_*`  functions to have a consistent api. The only other similar function is `detect_environment_specifier`. `detect_environment_specifier` will raise an error if no plugin is found, instead of returning `None`. I think either approach is fine, as long as the api is consistent. Maybe lets make this a follow up item to tackle once this PR is merged?